### ISSUE TITLE
HTTP Method filter for Breakpoints

### ISSUE THREAD

**Issue by xehpuk** [zaproxy/zaproxy#8542]
### Is your feature request related to a problem? Please describe.

I get too many "false positive" breaks because of CORS pre-flight (`OPTION`) requests.

### Describe the solution you'd like

An additional option to whitelist HTTP methods.

### Describe alternatives you've considered

None.

### Screenshots

Example:

!["Add Breakpoint" dialog with new option for HTTP Methods](/assets/3484906/15bf78bc-8c64-4bcc-9056-116a2d0084d9)

### Would you like to help fix this issue?

- [ ] Yes

**Comment by kingthorin** [comment]
Technically you could set the "Location" as Request Header and do it there via regex. But I agree it could probably be easier/nicer for users with a set of checkboxes or a multi-select combobox.

I couldn't quick find a site that had OPTIONS requests. But just as a PoC I used `^GET` as my pattern, with that on it broke on GETs once I checked "Inverse" it stopped catching GETs.

**Comment by xehpuk** [comment]
Ah, nice. So I can use `^(GET|POST) <url>`. Minor downside is that the URL should be escaped.

**Comment by kingthorin** [comment]
Yes or `^OPTIONS` and turn on Inverse. You probably don't even need to include `<url>` depends how you're working I guess.
